apps/meteor/app/livechat/server/hooks/processRoomAbandonment.js

```diff
@@ -64,7 +64,11 @@ callbacks.add(
 		if (!closedByAgent || !wasTheLastMessageSentByAgent) {
 			return;
 		}
-		const agentLastMessage = Messages.findAgentLastMessageByVisitorLastMessageTs(room._id, room.v.lastMessageTs);
+		// There's a possibility of a visitor not sending any message b4 agent, and this will cause
```

Comment:
Wouldn't the false condition here change the logic? From the variable name, it seems to be a "message from an agent", however, if this condition is false, then we're now picking the room's last message which isn't guaranteed to be a message from visitor

Reply:
You brought an interesting point. And I noticed this actually doesnt work (not the line, but the whole logic for selecting the last message). It _assumes_ the first message on a room will be _always_ a visitor message. Which makes sense because that's what livechat does (routing chat and do events only on first message)

However, if the message _after_ the visitor one is an app (like, what happens with CRMs) the query will return the message from the app, not from the agent.

So, I'll take some extra time to fix this behavior, or adapt the test to what currently happens and create a task for what may happen 👀